homeassistant/components/yeelight/binary_sensor.py

```diff
@@ -38,6 +39,14 @@ async def async_added_to_hass(self):
             )
         )
 
+    @property
+    def unique_id(self) -> Optional[str]:
+        """Return a unique ID."""
+        unique = self._device.unique_id
+
+        if unique:
+            return unique + "-nightlight_sensor"
+
```

Comment:
We should `return None` explicitly in the other case.

Reply:
Will do followup PR tomorrow with this.

Reply:
PR here: https://github.com/home-assistant/core/pull/35649